packages/babel-parser/src/parser/expression.js

```diff
@@ -504,7 +531,7 @@ export default class ExpressionParser extends LValParser {
     const startLoc = this.state.startLoc;
 
     return this.parseExprOp(
-      this.parseMaybeUnary(),
+      this.parseMaybeUnaryOrPrivate(),
```

Comment:
This change should also be applied to `parseFSharpPipelineBody` since the pipeline body descends to LogicalORExpression which includes private in. https://valtech-nyc.github.io/proposal-fsharp-pipelines/#prod-PipelineExpression

Reply:
It's done at https://github.com/babel/babel/pull/13727/files#diff-f876d1e822b0978b9c6de30b9fc6e96e2309c853a694feea274ecccf1d1d9f15R2935.